#This patch for DWM is meant to be applied after the alpha and xresources patches.
#currently only working with:
#colors
#borderpx, only applies to new windows

diff --git a/config.def.h b/config.def.h
index d765d7f..da5bb23 100644
--- a/config.def.h
+++ b/config.def.h
@@ -91,6 +91,7 @@ ResourcePref resources[] = {
 
 static const Key keys[] = {
 	/* modifier                     key        function        argument */
+	{ MODKEY,                       XK_r,      reload_xresources,{0} },
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
 	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
diff --git a/dwm.c b/dwm.c
index 367266a..3aa190f 100644
--- a/dwm.c
+++ b/dwm.c
@@ -254,6 +254,7 @@ static int xerrorstart(Display *dpy, XErrorEvent *ee);
 static void xinitvisual();
 static void zoom(const Arg *arg);
 static void load_xresources(void);
+static void reload_xresources(const Arg * arg);
 static void resource_load(XrmDatabase db, char *name, enum resource_type rtype, void *dst);
 
 /* variables */
@@ -2330,6 +2331,16 @@ load_xresources(void)
 	XCloseDisplay(display);
 }
 
+void
+reload_xresources(const Arg * arg)
+{
+	load_xresources();
+	for (int i = 0; i < LENGTH(colors); i++)
+		scheme[i] = drw_scm_create(drw, colors[i], alphas[i], 3);
+	focus(NULL);
+	arrange(NULL);
+}
+
 int
 main(int argc, char *argv[])
 {
